source/common/tracing/http_tracer_manager_impl.cc

```diff
@@ -18,11 +18,14 @@ HttpTracerManagerImpl::getOrCreateHttpTracer(const envoy::config::trace::v3::Tra
   const auto cache_key = MessageUtil::hash(*config);
   const auto it = http_tracers_.find(cache_key);
   if (it != http_tracers_.end()) {
-    return it->second;
+    auto http_tracer = it->second.lock();
+    if (http_tracer) { // HttpTracer might have been released since it's a weak reference
+      return http_tracer;
+    }
```

Comment:
The weak references are still potentially never going to leave the map so this can still technically leak. You will need some sweeper to actually clean this up. Given the likely small number of trace configs and also the small number of times this is called, it might be sufficient to just iterate through all cached configs and remove any that can't be locked?

Reply:
Fixed